### ISSUE TITLE
[google_maps]: onCameraMoveStarted stops working after tap outside map view

### ISSUE THREAD

**Issue by EugenePisotsky** [flutter/flutter#35323]
Hello! Here's an example:

```dart
import 'package:flutter/material.dart';
import 'package:google_maps_flutter/google_maps_flutter.dart';

class TestRoute extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Stack(
      children: <Widget>[
        GoogleMap(
          initialCameraPosition: CameraPosition(
            target: LatLng(36.536123, -86.671607),
            zoom: 16
          ),
          onCameraMoveStarted: () {
            print('start moving');
          },
        ),
        Positioned(
          top: 0,
          left: 0,
          right: 0,
          child: Container(
            height: 100,
            color: Colors.white,
          ),
        )
      ],
    );
  }
}
```

`onCameraMoveStarted` callback works until I tap on the white container, after I do it it's called only once and not called anymore. The same problem with `onCameraIdle`, but `onCameraMove` works fine.

I tried only iOS, not sure if problem exists on Android too.

<details> 
<summary>flutter doctor -v</summary> 

```console
[✓] Flutter (Channel unknown, v1.7.12-pre.40, on Mac OS X 10.14.5 18F132, locale
    en-UA)
    • Flutter version 1.7.12-pre.40 at /Users/eugenepisotsky/Development/flutter
    • Framework revision 3badcf51a4 (12 hours ago), 2019-06-28 15:14:03 -0700
    • Engine revision e96900df2f
    • Dart version 2.4.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/eugenepisotsky/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling
      support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_152-release-1343-b01)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 10.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.2, Build version 10E125
    • CocoaPods version 1.7.2

[✓] iOS tools - develop for iOS devices
    • ios-deploy 1.9.4

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.4)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 35.3.1
    • Dart plugin version 183.6270
    • Java version OpenJDK Runtime Environment (build
      1.8.0_152-release-1343-b01)

[✓] IntelliJ IDEA Community Edition (version 2019.1.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 35.3.3
    • Dart plugin version 191.7221

[✓] Connected device (3 available)
    • iPhone (Eugene) • 1f7899e1777a61e4f950a866195f88b96e420b85 • ios
      • iOS 12.3.1
    • macOS           • macOS                                    • darwin-x64
      • Mac OS X 10.14.5 18F132
    • Chrome          • chrome                                   •
      web-javascript • Google Chrome 75.0.3770.80 


• No issues found!
```
	
</details>

**Comment by papalardo** [comment]
I have a same problem.

**Comment by BondarenkoStas** [comment]
@EugenePisotsky Is it the same while using current flutter stable and google_maps_flutter 0.5.21+7?

**Comment by iapicca** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. We are therefore reluctantly going to close this bug for now. Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!
  
Thanks for your contribution.

**Comment by EugenePisotsky** [comment]
> @EugenePisotsky Is it the same while using current flutter stable and google_maps_flutter 0.5.21+7?

@BondarenkoStas I tried 0.5.21+9, still the same. more information:

1. on android it works as expected
2. on iOS, when I click on button, everything is fine (I think it works fine with all gesture detector widgets), but when I click on simple view, it stops working

**Comment by iapicca** [comment]
Hi @EugenePisotsky 
can you please provide your updated  `flutter doctor -v`
and your `flutter run --verbose`?
Thank you

**Comment by VladyslavBondarenko** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. We are therefore reluctantly going to close this bug for now. Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.
Could everyone who still has this problem please file a new issue with the exact descriptions what happens, logs and the output of 'flutter doctor -v'  please.
All system setups can be slightly different so its always better to open new issues and reference related issues.

**Comment by renanyoy** [comment]
I have same problem, works well on android and not on iOS, onCameraMoveStarted doesn't work all times

**Comment by renanyoy** [comment]
I have a search bar in overlay it seems it happens when the search bar has the keyboard focus, but not each time..

**Comment by bluelet13** [comment]
I had the same problem.
Call to `moveCamera` or `animateCamera` when tapping anything other than the map, seemed to get workaround the issue for me.

```
final level = await controller.getZoomLevel();
await controller.moveCamera(CameraUpdate.zoomTo(level));
```

**Comment by renanyoy** [comment]
@bluelet13 I have a complex UI on top of the map, not easy to implement this workaround for me

**Comment by magax** [comment]
I have a button in a stack on top of the map that is supposed to display off itself on tap, then I want to display the button back when the camera starts moving but I'm facing the same issue. Only OnCameraMove works but I'm not sure if it's very good to use it for that feature, it fires so many times.

**Comment by TahaTesser** [comment]
Hi @EugenePisotsky 
Using the code sample I can't reproduce the issue on the latest `google_maps_flutter` 


https://user-images.githubusercontent.com/48603081/115714540-e4384880-a37f-11eb-96c4-20a8839da37f.mov

<details> 
<summary>complete code sample</summary> 

```dart
import 'package:flutter/material.dart';
import 'package:google_maps_flutter/google_maps_flutter.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: TestRoute(),
    );
  }
}

class TestRoute extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Stack(
      children: <Widget>[
        GoogleMap(
          initialCameraPosition:
              CameraPosition(target: LatLng(36.536123, -86.671607), zoom: 16),
          onCameraMoveStarted: () {
            print('start moving');
          },
        ),
        Positioned(
          top: 0,
          left: 0,
          right: 0,
          child: Container(
            height: 100,
            color: Colors.red,
          ),
        )
      ],
    );
  }
}

``` 

</details>


```yaml
dependencies:
  flutter:
    sdk: flutter
  cupertino_icons: ^0.1.3
  google_maps_flutter: ^2.0.3
  google_maps_flutter_web: ^0.3.0
```

<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel stable, 2.0.5, on macOS 11.2.3 20D91 darwin-x64, locale
    en-GB)
    • Flutter version 2.0.5 at /Users/tahatesser/Code/flutter_stable
    • Framework revision adc687823a (6 days ago), 2021-04-16 09:40:20 -0700
    • Engine revision b09f014e96
    • Dart version 2.12.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.55.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (3 available)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E • ios            • iOS
      14.4.2
    • macOS (desktop)        • macos                     • darwin-x64     •
      macOS 11.2.3 20D91 darwin-x64
    • Chrome (web)           • chrome                    • web-javascript •
      Google Chrome 90.0.4430.85

• No issues found!
```

```bash
[✓] Flutter (Channel master, 2.2.0-11.0.pre.215, on macOS 11.2.3 20D91
    darwin-x64, locale en-GB)
    • Flutter version 2.2.0-11.0.pre.215 at
      /Users/tahatesser/Code/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5865c14d67 (7 hours ago), 2021-04-22 06:44:03 +0200
    • Engine revision 7e69744ede
    • Dart version 2.14.0 (build 2.14.0-18.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Volumes/Extreme/SDK
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[!] IntelliJ IDEA Community Edition (version unknown)
    ✗ Invalid IntelliJ version number.

[✓] VS Code (version 1.55.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (3 available)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E • ios            • iOS
      14.4.2
    • macOS (desktop)        • macos                     • darwin-x64     •
      macOS 11.2.3 20D91 darwin-x64
    • Chrome (web)           • chrome                    • web-javascript •
      Google Chrome 90.0.4430.85

! Doctor found issues in 1 category.
```
	
</details>

Closing, as this isn't an issue with `google_maps_flutter`  itself. If you disagree, please write in the comments and I will reopen it.
Thank you
